lib/routemap.c

```diff
@@ -1663,13 +1676,15 @@ static bool route_map_dep_hash_cmp(const void *p1, const void *p2)
 static void route_map_clear_reference(struct hash_bucket *bucket, void *arg)
 {
 	struct route_map_dep *dep = (struct route_map_dep *)bucket->data;
-	char *rmap_name;
+	struct route_map_dep_data *dep_data = NULL;
 
 	if (arg) {
-		rmap_name =
-			(char *)hash_release(dep->dep_rmap_hash, (void *)arg);
-		if (rmap_name) {
-			XFREE(MTYPE_ROUTE_MAP_NAME, rmap_name);
+		dep_data = (struct route_map_dep_data *)hash_release(
```

Comment:
we do not need to cast a void pointer here

Reply:
Thanks, @donaldsharp. I've addressed this.